Add Stryker4s sbt performance blog (#292)
stryker-mutator/stryker-mutator.github.io: blog/2021-06-16-stryker4s-sbt-performance.mdx

```

We now know what tests are in your project. Let's have a look at running those tests:

### 🧪 Meet: test interfaces

This is something our friends in the NodeJS world are (rightfully) a little [jealous of](https://stryker-mutator.io/docs/stryker-js/plugins#test-runners) 😏. When you call `sbt test`, sbt will look for any registered test frameworks that implement the [test-interface](https://github.com/sbt/test-interface) (from `loadedTestFrameworks`). This is a small collection of interfaces that let a test framework tell sbt how to run tests in that framework. Because all test frameworks in sbt implement this interface, we have a single programmatic way to call any test framework. It doesn't matter if you're using ScalaTest, MUnit, Weaver-test or anything else. Using a bit of reflection, we can create an instance of the test framework runner and run tests:

```scala
val framework = classLoader.loadClass(testFrameworkClass).getConstructor().newInstance().asInstanceOf[Framework]
val runner = framework.runner(args, remoteArgs, classLoader)

val testTasks: Array[Task] = runner.tasks(testGrouping.taskDefs)

testTasks.foreach(testTask => {
  testTask.execute()
})

> ```

Reviewer: I think a line here explaining how this test interface is used in Stryker4s would be helpful.

### 🔀 Meet: concurrency

> Stryker4s activates each mutant separately and runs your unit tests against _only_ the change that mutant creates. This is to make sure one mutant won't have an impact on the result of another. However, there's no rule that says we can't start multiple Java processes at the same time and run those in parallel.

Reviewer: ```suggestion
Stryker4s activates and runs each mutant one by one in isolation. This is to make sure one mutant won't have an impact on the result of another. This is done by creating a separate process for it. However, there's no rule that says we can't start multiple Java processes at the same time and run those in parallel.
```
